uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -0,0 +1,485 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components'
+], function (_, moment, components) {
+  components.component('leaveCalendarMonth', {
+    bindings: {
+      contacts: '<',
+      month: '<',
+      period: '<',
+      showContactName: '<',
+      showOnlyWithLeaveRequests: '<',
+      supportData: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-calendar-month.html';
+    }],
+    controllerAs: 'month',
+    controller: ['$log', '$q', '$rootScope', 'Calendar', 'LeaveRequest', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, Calendar, LeaveRequest, sharedSettings) {
+    $log.debug('Component: leave-calendar-month');
+
+    var dataLoaded = false;
+    var calendars = {};
+    var leaveRequests = {};
+
+    var vm = this;
+    vm.currentPage = 0;
+    vm.pageSize = 20;
+    vm.visible = false;
+    vm.showContactName = vm.showContactName ? !!vm.showContactName : false;
+    vm.showOnlyWithLeaveRequests = vm.showOnlyWithLeaveRequests ? !!vm.showOnlyWithLeaveRequests : false;
+
+    vm.contactsList = contactsList;
+
+    (function init () {
+      var dateFromMonth = moment().month(vm.month.index).year(vm.month.year);
+
+      indexData();
+      initListeners();
+
+      vm.month = buildMonthStructure(dateFromMonth);
+
+      $rootScope.$emit('LeaveCalendar::monthInjected');
+    }());
+
+    /**
+     * Adds a leave request to the calendar
+     *
+     * @param {Object} event
+     * @param {LeaveRequestInstance} leaveRequest
+     */
+    function addLeaveRequest (event, leaveRequest) {
```

Comment:
Using `__` to indicate unused params (maybe I should add this to the styleguide.

It is common to use `_` in JS/Ruby (https://stackoverflow.com/questions/32197927/standard-conventions-for-indicating-a-function-argument-is-unused-in-javascript), but given we have `lodash` then the double underscore avoids conflicts